air/src/proof/ood_frame.rs

```diff
@@ -42,14 +47,12 @@ impl OodFrame {
     ///
     /// # Panics
     /// Panics if evaluation frame has already been set.
-    pub fn set_trace_states<E: FieldElement>(&mut self, trace_states: &[Vec<E>]) {
+    pub fn set_trace_states<E: FieldElement>(&mut self, trace_states: Vec<E>) {
         assert!(
             self.trace_states.is_empty(),
             "trace sates have already been set"
         );
-        for trace_state in trace_states {
-            trace_state.write_into(&mut self.trace_states);
-        }
+        trace_states.write_into(&mut self.trace_states);
```

Comment:
I wonder if we should keep the old interface and handle interleaved serialization here instead. We could also return the interleaved result from here so that in the prover channel we could just call `reseed()` with it. So, the signature could look something like this:
```Rust
pub fn set_trace_states<E: FieldElement>(&mut self, trace_states: &[Vec<E>]) -> Vec<E>
```

Also, maybe we should serialize the frame size as the first value - this way, we won't need to rely on `FRAME_SIZE` constant.

Reply:
Makes sense. I have changed it accordingly.